meetings/2022-05-02.md

```diff
@@ -0,0 +1,118 @@
+* AvdB: let me know if there is anything I can do to help with this
+* AC: There are so many permutations since there are so many contexts where clients try to authenticate. Give all of them we can't always say that client MUST do it this way. We are also trying to move ecosystemm along in a step by step fashion. If we don't allow other mechanisms than we get into compatibility issues
+* eP: I'd like to consider making Client Identifiers required for an OP, otherwise it's not Solid compliant. That would also give more confidence by the client.
+* AC: It would be helpful to add paragraph that states that OP MUST support Client ID documents.
+* ...: There may be ambiguity with regards what is and what isn't dereferencable
+
+ACTION: AC to clarify requirements for OP support of Client ID documents
+
+* AvdB: is there a case where the `client_id` value is dereferencable but _not_ a Client Id Document?
+* AC: I think there are 2 parts. 1. If it is dereferencable, if it is 2. Fetch the document and check if it is `application/ld+json`, it must be valid JSON, it also needs to use the normative `@context`.
+* ...: Let's say we have some other specification which also defines dereferencable client identifiers, not JSON-LD. When OP tries to dereference it, it would need to have condition based on the format of the client id document. Even if both are JSON-LD it could check for the `@context`.
+* AvdB: if it is required that an OP is required to dereference these client_id URIs, then it makes interop much easier to achieve.
+* eP: the only case when it's not a client identifier, it's an identifier issued by the OP, which will be pre-registered by the OP. The OP can then just check if it's one of these other formats, it can use that; otherwise, it can just dereference the URI as a client identifier.
+* AC: This is how I implemented it.
+* eP: even for static registration, an OP will still manage these identifiers
+* AC: This is exactly my experience with implementation. The identifier is either one that was created by the OP, or an external one which would be a Client ID IRI.
+
+### UMA updates
+
+* LD: I am working with the CSS project, having trouble with DPoP during the claims pushing stage. This is unclear on the Solid-OIDC side. This could be an opportunity to specify an IRI for a DPoP-bound ID token.
+* AC: The way I've done this, in one stage you may be providing ID Token. When you push that claim it may perform validation of that ID Token. Also part of this validation is validating the DPOP binding. The part of this interaction involves DPoP header. The format is definitelly OIDC ID Token with some extra claims. One could argue that it is ID Token but also an argument could be made that it is a different claim format.
```

Comment:
```suggestion
* AC: The way I've done this, in one stage you may be providing ID Token. When you push that claim, it may perform validation of that ID Token. Also part of this validation is validating the DPOP binding. This part of this interaction involves the DPoP header. The format is definitely OIDC ID Token with some extra claims. One could argue that it is ID Token, but an argument could also be made that it is a different claim format.
```